src/components/Subplaylist.tsx

```diff
@@ -54,6 +93,77 @@ export default function Subplaylist (props: {
   const classes = useStyles()
   const [selectedGenres, setSelectedGenres] = useState<string[]>([])
 
+  const [sliders, setSliders] = useState<{ name: string, min: number, max: number }[]>([]);
+
+  const deleteSlider = (id: String) => {
+    console.log("Deleting slider ", id);
+    setSliders(sliders.filter(k => k.name !== id));
+  }
+
+  const updateSlider = (id: String, range: Number[]) => {
+    setSliders(
+      sliders.map(
+        el => el.name === id ? { ...el, min: Number(range[0]), max: Number(range[1]) } : el
+      )
+    )
+
+  }
+  const getFeaturesFromSlider = (name: string, incomingFeatures: Number[]) => {
+    updateSlider(name, incomingFeatures)
+
+  }
+
+  const getOptionFromMenu = (option: any) => {
+    var found = false;
+    for (var i = 0; i < sliders.length; i++) {
+      if (sliders[i].name === option.name) {
+        found = true;
+        break;
+      }
+    }
+    if (!found) {
+      setSliders([...sliders, option])
+    }
+  }
+  const TrackInRange = (track: TrackObj): boolean => {
+    var found = true;
+    sliders.map((slider) => {
+      if (track.features) {
+        if (slider.name === 'Acousticness' && (track.features.acousticness < slider.min / 100 || track.features.acousticness > slider.max / 100)) {
+          found = false;
+        }
+        if (slider.name === 'Danceability' && (track.features.danceability < slider.min / 100 || track.features.danceability > slider.max / 100)) {
+          found = false;
+        }
+        if (slider.name === 'Energy' && (track.features.energy < slider.min / 100 || track.features.energy > slider.max / 100)) {
+          found = false;
+        }
+        if (slider.name === 'Instrumentalness' && (track.features.instrumentalness < slider.min / 100 || track.features.instrumentalness > slider.max / 100)) {
+          found = false;
+        }
+        if (slider.name === 'Liveness' && (track.features.liveness < slider.min / 100 || track.features.liveness > slider.max / 100)) {
+          found = false;
+        }
+        // if (slider.name === 'Loudness' && track.features.loudness < slider.min/100 || track.features.loudness > slider.max/100){
+        //   found=false;
+        // }
+        if (slider.name === 'Speechiness' && (track.features.speechiness < slider.min / 100 || track.features.speechiness > slider.max / 100)) {
+          found = false;
+        }
+        // if (slider.name === 'Tempo' && track.features.tempo < slider.min/100 || track.features.tempo > slider.max/100){
+        //   found=false;
+        // }
+        if (slider.name === 'Valence' && (track.features.valence < slider.min / 100 || track.features.valence > slider.max / 100)) {
+          found = false;
+        }
+      }
```

Comment:
Can this be refactored along with the other feature code sections to function dynamically

Reply:
What exactly do you mean by function dynamically here?

Reply:
You can see here that the code is repeated, differing only by the lookup name, and track property access. This can be refactored to simplify maintainability